text/0156-columnar-encoding.md

```diff
@@ -65,6 +65,192 @@ we accelerate the development of the OpenTelemetry protocol while expanding its
+        ))
+    ), true),
+
+    // exemplars
+    Field::new("exemplars", DataType::List(
+        Box::new(Field::new(
+            "exemplar",
+            DataType::Struct(vec![
+                Field::new("filtered_attributes", DataType::List(
+                    Box::new(Field::new(
+                        "attribute",
+                        DataType::Struct(vec![
+                            Field::new("name", DataType::Utf8, false),
+                            Field::new("value", DataType::Utf8, false),
+                        ]),
+                        true,
+                    ))
+                ), true),
+                Field::new("filtered_labels", DataType::List(
+                    Box::new(Field::new(
+                        "label",
+                        DataType::Struct(vec![
+                            Field::new("name", DataType::Utf8, false),
+                            Field::new("value", DataType::Utf8, false),
+                        ]),
+                        true,
+                    ))
+                ), true),
+                Field::new("time_unix_nano", DataType::UInt64, false),
+                // Could be Float64 or Int64
+                Field::new("value", DataType::Float64, false),
+                Field::new("span_id", DataType::Binary, false),
+                Field::new("trace_id", DataType::Binary, false),
+            ]),
+            true,
+        ))
+    ), true),
+])
+```
+
+### OpenTelemetry logs to Arrow mapping
+
+```rust
+// log schema declaration
+Schema::new(vec![
+    // time
+    Field::new("time_unix_nano", DataType::UInt64, false),
+
+    Field::new("severity_number", DataType::UInt8, false),
+    Field::new("severity_text", DataType::Utf8, false),
+    Field::new("name", DataType::Utf8, false),
+    Field::new("body", DataType::Utf8, false),
+
+    // attributes
+    Field::new("attributes", DataType::List(
+        Box::new(Field::new(
+            "attribute",
+            DataType::Struct(vec![
+                Field::new("name", DataType::Utf8, false),
+                Field::new("value", DataType::Utf8, false),
+            ]),
+            true,
+        ))
+    ), true),
+
+    Field::new("flags", DataType::Int32, false),
+    Field::new("span_id", DataType::Binary, false),
+    Field::new("trace_id", DataType::Binary, false),
+])
+```
+
+### OpenTelemetry traces to Arrow mapping
+
+```rust
+// trace schema declaration
+Schema::new(vec![
+    // times
+    Field::new("start_time_unix_nano", DataType::UInt64, false),
+    Field::new("end_time_unix_nano", DataType::UInt64, false),
+
+    Field::new("trace_id", DataType::Binary, false),
+    Field::new("span_id", DataType::Binary, false),
+    Field::new("trace_state", DataType::Utf8, false),
+    Field::new("parent_span_id", DataType::Binary, false),
+    Field::new("name", DataType::Utf8, false),
+    Field::new("kind", DataType::UInt8, false),
+
+    // attributes
+    Field::new("attributes", DataType::List(
```

Comment:
Is there some pointer to Arrow documentation that describes the memory layout of nested fields like that?

Reply:
Yes. Let me complete the document with some additional details and a description of the options we have (i.e. a list of structs or a struct flattening all the attributes in independent columns).

Reply:
Added a link to the documentation.

Reply:
Ref to link location: https://github.com/open-telemetry/oteps/pull/171/files#diff-8b2a4811d420878fd569924408084d8731e0915142a33cc1bfeb1242441247c1R243